graph-tool: 2.16 -> 2.26

ciderale opened the pull request:
###### Motivation for this change

Reenable support for graph-tool in python3 environments and update to newest version

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [X] other Linux distributions (inside Docker container)
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Details:
- add ncurses: configure links against ncurses and fails otherwise
    configure: error: Could not link test program to Python.
    https://travis-ci.org/NixOS/nixpkgs/builds/48759067
    The given hint (Maybe the main Python library has been installed
    in some non-standard library path) is misleading.
    The config.log reveals that the failure is due to missing ncurses link option
- with-boost-libdir is need to find Boost::IOStreams/regex/etc.
- expat/cgal are detected in /usr/lib when not specified explicitly
- boost > boost159 is needed to have -lboost_python3 (and -lboost_python)

- set pythonModule = Python;
  => inorder to be used in python.buildEnv { extraLibs = [..]; }

tested on MacOSX and in a linux Docker container with:
```
> nix-shell -I nixpkgs=. -p python2.pkgs.graph-tool
> nix-shell -I nixpkgs=. -p python3.pkgs.graph-tool
and created a simple graph
>>> import graph_tool
>>> from graph_tool.all import *
>>> g = Graph()
>>> v1 = g.add_vertex()
>>> graph_draw(g, vertex_text=g.vertex_index, vertex_font_size=18,output="one-nodes.png")
```
---

ciderale committed:
graph-tool: 2.16 -> 2.26

- add ncurses: configure links against ncurses and fails otherwise
    configure: error: Could not link test program to Python.
    https://travis-ci.org/NixOS/nixpkgs/builds/48759067
    The given hint (Maybe the main Python library has been installed
    in some non-standard library path) is misleading.
    The config.log reveals that the failure is due to missing ncurses link option
- with-boost-libdir is need to find Boost::IOStreams/regex/etc.
- expat/cgal are detected in /usr/lib when not specified explicitly
- boost > boost159 is needed to have -lboost_python3 (and -lboost_python)

- set pythonModule = Python;
  => inorder to be used in python.buildEnv { extraLibs = [..]; }

tested on MacOSX and in a linux Docker container with:
> nix-shell -I nixpkgs=. -p python2.pkgs.graph-tool
> nix-shell -I nixpkgs=. -p python3.pkgs.graph-tool

FRidh commented on pkgs/development/python-modules/graph-tool/2.x.x.nix:
```diff
@@ -1,11 +1,13 @@
+  version = "2.26";
   name = "${python.libPrefix}-graph-tool-${version}";
 
+  pythonModule = python;
```
it's better to use `toPythonModule (stdenv.mkDerivation {...})`

FRidh replied:
Even better would be to try and use `buildPythonPackage` with `format = "other";`

ciderale replied:
Thanks a lot for the quick feedback. I did try `buildPythonPackage` earlier, but I did not know about `format = "other"`. With the additional `format` attribute it works. I only retested on OSX/Py3k tough as the compilation takes ages and I don't think that change will have an effect on the other combination, or will it?

ciderale committed:
graph-tool: use buildPythonPackage with format = "other";

ciderale commented:
@FRidh Thanks again for your feedback. I think I addressed your comment, but I'm not sure how if I have to press some button to mark the "Change Requested" as done, or if commenting is enough. Let me know if there are still some open points.

dotlambda commented:
@matthewbauer Did you read https://groups.google.com/forum/#!topic/nix-devel/IA6o_CRI0NY?
